amethyst_ui/src/button/builder.rs

```diff
@@ -53,6 +50,12 @@ pub struct UiButtonBuilder {
     font_size: f32,
     image: Option<TextureHandle>,
     parent: Option<Entity>,
+    hover_image: Option<TextureHandle>,
+    hover_text_color: Option<[f32; 4]>,
+    press_image: Option<TextureHandle>,
+    press_text_color: Option<[f32; 4]>,
+    hover_sound: Option<SourceHandle>,
+    press_sound: Option<SourceHandle>,
```

Comment:
This is what I said I don't like. The separation of concern isn't there. You have audio data at the same place as visual and layouting. While I understand where you are coming from, I still think it would be better to have separated components than a big struct of Option.

When we'll be doing other elements (checkbox for example), you'll need to copy paste the data for all of those in the checkbox builder.

Tell me if I'm not being clear with what I'm trying to say.